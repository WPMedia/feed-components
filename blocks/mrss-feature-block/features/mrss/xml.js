import PropTypes from 'fusion:prop-types'
import Consumer from 'fusion:consumer'
import get from 'lodash/get'
import moment from 'moment'
import getProperties from 'fusion:properties'
import { resizerKey } from 'fusion:environment'
import { BuildContent } from '@wpmedia/feeds-content-elements'
import { generatePropsForFeed } from '@wpmedia/feeds-prop-types'
import { buildResizerURL } from '@wpmedia/feeds-resizer'
import { findVideo } from '@wpmedia/feeds-find-video-stream'

const jmespath = require('jmespath')

const rssTemplate = (
  elements,
  {
    channelTitle,
    channelDescription,
    channelPath,
    channelCopyright,
    channelTTL,
    channelUpdatePeriod,
    channelUpdateFrequency,
    channelCategory,
    channelLogo,
    imageTitle,
    imageCaption,
    imageCredits,
    itemTitle,
    itemDescription,
    pubDate,
    itemCredits,
    itemCategory,
    includeContent,
    resizerURL,
    domain,
    feedTitle,
    feedLanguage,
    videoInfo,
    mrssBuildContent,
  },
) => ({
  rss: {
    '@xmlns:atom': 'http://www.w3.org/2005/Atom',
    '@version': '2.0',
    '@xmlns:media': 'http://search.yahoo.com/mrss/',
    channel: {
      title: `${channelTitle || feedTitle + ' Videos'}`,
      link: domain,
      'atom:link': {
        '@href': `${domain}${channelPath}`,
        '@rel': 'self',
        '@type': 'application/rss+xml',
      },
      description: channelDescription || feedTitle,
      pubDate: moment.utc(Date.now()).format('ddd, DD MMM YYYY HH:mm:ss ZZ'),
      ...(channelLogo && {
        image: {
          url: buildResizerURL(channelLogo, resizerKey, resizerURL),
          title: `${channelTitle || feedTitle + ' Videos'}`,
          link: domain,
        },
      }),

      item: elements.map((s) => {
        const url = `${domain}${s.website_url || s.canonical_url}`
        const img =
          s.promo_items && (s.promo_items.basic || s.promo_items.lead_art)
        const videoStream = findVideo(s, videoInfo)

        return {
          title: `${jmespath.search(s, itemTitle)}`,
          link: url,
          ...(itemCredits &&
            (jmespath.search(img, itemCredits) || []).length && {
              'media:credit': {
                '@role': 'producer',
                $: jmespath.search(img, itemCredits).join(','),
              },
            }),
          ...(itemDescription && {
            description: jmespath.search(s, itemDescription),
          }),
          guid: {
            '@isPermaLink': false,
            '#': url,
          },
          referenceid: s._id,
          pubDate: moment
            .utc(s[pubDate])
            .format('ddd, DD MMM YYYY HH:mm:ss ZZ'),

          'media:content': {
            '@': {
              isDefault: 'true',
              ...(s.duration && {
                duration: Math.trunc(s.duration / 1000),
              }),
              ...(videoStream && {
                ...(videoInfo.url && { url: videoInfo.url }),
                ...(videoInfo.height && { height: videoInfo.height }),
                ...(videoInfo.width && { width: videoInfo.width }),
                ...(videoInfo.bitrate && { bitrate: videoInfo.bitrate }),
                ...(videoInfo.stream_type && {
                  type: videoInfo.stream_type,
                  '#': videoStream,
                }),
              }),
            },
          },
          'media:keywords': (
            jmespath.search(s, 'taxonomy.seo_keywords[*]') || []
          ).join(','),
          ...(s.description ||
            (s.subheadlines &&
              s.subheadlines.basic && {
                'media:caption': {
                  $: s.description || s.subheadlines.basic,
                },
              })),
          'media:transcript': s.transcript,
          ...(s.taxonomy &&
            s.taxonomy.primary_section &&
            s.taxonomy.primary_section.name && {
              'media:category': s.taxonomy.primary_section.name,
            }),
          ...(img &&
            img.url && {
              'media:thumbnail': {
                '@url': buildResizerURL(img.url, resizerKey, resizerURL),
              },
            }),
        }
      }),
    },
  },
})

export function Mrss({ globalContent, customFields, arcSite }) {
  const {
    resizerURL = '',
    feedDomainURL = '',
    feedTitle = '',
    feedLanguage = '',
  } = getProperties(arcSite)

  function MrssBuildContent() {
    BuildContent.call(this)
  }
  const mrssBuildContent = new MrssBuildContent()

  // can't return null for xml return type, must return valid xml template
  return rssTemplate(get(globalContent, 'content_elements', []), {
    ...customFields,
    resizerURL,
    domain: feedDomainURL,
    feedTitle,
    feedLanguage,
    mrssBuildContent,
  })
}

Mrss.propTypes = {
  customFields: PropTypes.shape({
    channelPath: PropTypes.string.tag({
      label: 'Path',
      group: 'Channel',
      description:
        'Path to the feed, excluding the domain, defaults to /arcio/mrss',
      defaultValue: '/arcio/mrss/',
    }),
    selectVideo: PropTypes.kvp.tag({
      label: 'Select video using',
      group: 'Video',
      description:
        'This criteria is used to filter videos encoded in the streams array',
      defaultValue: {
        height: '',
        width: '',
        bitrate: 5400,
        stream_type: 'mp4',
      },
    }),
    ...generatePropsForFeed('rss', PropTypes),
  }),
}
Mrss.label = 'MRSS'
export default Consumer(Mrss)
