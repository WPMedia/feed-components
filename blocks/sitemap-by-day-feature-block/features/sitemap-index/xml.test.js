// eslint-disable-next-line no-unused-vars
import Consumer from 'fusion:consumer'
import { SitemapByDay } from './xml'

it('returns template with default values', () => {
  const sitemapbyday = SitemapByDay({
    arcSite: 'the-globe',
    globalContent: {
      count: 437,
      content_elements: [
        {
          last_updated_date: '2020-04-07T15:02:08.918Z',
          website_url: '/food/2020/04/07/tips-for-safe-hand-washing',
        },
      ],
    },
    customFields: {
      feedPath: '/arc/outboundfeeds/sitemap',
      feedName: '/sitemap-by-day/',
      feedParam: '&outputType=xml',
      lastMod: 'last_updated_date',
    },
    requestUri: '/sitemap-by-day/2021-04-06/?outputType=xml',
  })
  expect(sitemapbyday).toMatchSnapshot()
})

it('returns template with category in url', () => {
  const sitemapbyday = SitemapByDay({
    arcSite: 'the-globe',
    globalContent: {
      count: 437,
      content_elements: [
        {
          last_updated_date: '2020-04-07T15:02:08.918Z',
          website_url: '/food/2020/04/07/tips-for-safe-hand-washing',
        },
      ],
    },
    customFields: {
      feedPath: '/arc/outboundfeeds/sitemap',
      feedName: '/sitemap-by-day/',
      feedParam: '&outputType=xml',
      lastMod: 'last_updated_date',
    },
    requestUri:
      '/arc/outboundfeeds/sitemap-by-day/category/sports/2021-04-06?outputType=xml',
  })
  expect(sitemapbyday).toMatchSnapshot()
})

it('returns template with 10000 results ES 7 cap', () => {
  const sitemapbyday = SitemapByDay({
    arcSite: 'the-globe',
    globalContent: {
      count: 10000,
      content_elements: [
        {
          last_updated_date: '2020-04-07T15:02:08.918Z',
          website_url: '/food/2020/04/07/tips-for-safe-hand-washing',
        },
      ],
    },
    customFields: {
      feedPath: '/arc/outboundfeeds/sitemap',
      feedName: '/sitemap-by-day/',
      feedParam: '&outputType=xml',
      lastMod: 'last_updated_date',
    },
    requestUri:
      '/arc/outboundfeeds/sitemap-index/category/sports?outputType=xml',
  })
  expect(sitemapbyday).toMatchSnapshot()
})

it('returns template with no content', () => {
  const sitemapbyday = SitemapByDay({
    arcSite: 'the-globe',
    globalContent: {},
    customFields: {
      feedPath: '/arc/outboundfeeds/sitemap',
      feedName: '/sitemap-by-day/',
      feedParam: '&outputType=xml',
      lastMod: 'last_updated_date',
    },
    requestUri:
      '/arc/outboundfeeds/sitemap-index/category/sports?outputType=xml',
  })
  expect(sitemapbyday).toMatchSnapshot()
})
