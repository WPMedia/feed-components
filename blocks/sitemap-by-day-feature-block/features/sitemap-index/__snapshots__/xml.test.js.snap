// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns template with 10000 results ES 7 cap 1`] = `
Object {
  "sitemapindex": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "sitemap": Array [
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=0&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=1900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=2900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=3900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=4900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=5900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=6900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=7900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=8900&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9000&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9400&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9500&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9600&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9700&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9800&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap?from=9900&outputType=xml",
      },
    ],
  },
}
`;

exports[`returns template with category in url 1`] = `
Object {
  "sitemapindex": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "sitemap": Array [
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemapcategory/sports/2021-04-06?from=0&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemapcategory/sports/2021-04-06?from=100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemapcategory/sports/2021-04-06?from=200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemapcategory/sports/2021-04-06?from=300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemapcategory/sports/2021-04-06?from=400&outputType=xml",
      },
    ],
  },
}
`;

exports[`returns template with default values 1`] = `
Object {
  "sitemapindex": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "sitemap": Array [
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap2021-04-06/?from=0&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap2021-04-06/?from=100&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap2021-04-06/?from=200&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap2021-04-06/?from=300&outputType=xml",
      },
      Object {
        "lastmod": "2020-04-07T15:02:08.918Z",
        "loc": "http://demo-prod.origin.arcpublishing.com/arc/outboundfeeds/sitemap2021-04-06/?from=400&outputType=xml",
      },
    ],
  },
}
`;

exports[`returns template with no content 1`] = `
Object {
  "sitemapindex": Object {
    "@xmlns": "http://www.sitemaps.org/schemas/sitemap/0.9",
    "sitemap": Array [],
  },
}
`;
